### PR TITLE
Bug 1907373: Rebase to kube v1.20.0

### PR DESCRIPTION
**Pull request by tnozicka**
tracking sheet https://docs.google.com/spreadsheets/d/10KYptJkDB1z8_RYCQVBYDjdTlRfyoXILMa0Fg8tnNlY

TODO:
- [x] Update upstream tags on o/k repo (https://github.com/openshift/kubernetes/tree/v1.20.0)
- [x] Update openshift/api repository to v1.20.0 (openshift/api#789)
- [x] Update openshift/client-go to v1.20.0 (openshift/client-go#172)
- [x] Update openshift/library-go to v1.20.0 (openshift/library-go#942)
- [ ] Update openshift/apiserver-library-go to v1.20.0 (openshift/apiserver-library-go#39)
- [x] Update the hyperkube image version to the release tag
- [ ] Bump updated openshift dependencies
- [x] Update etcd data locations in origin tests (https://github.com/openshift/origin/pull/25717)
- [x] Cluster quota fix for new CM in every namespace (https://github.com/openshift/origin/pull/25730)
- [x] Test-cmd error message fix (https://github.com/openshift/origin/pull/25731)
- [ ] Update REBASE.openshift.md
- [ ] Squash commits

Followups:
- [ ] Revendor to openshift/origin ()

EDIT: it has been rebased onto v1.20.0 from the initial v1.20.0-beta.2

/cc @soltysh @sttts

### CODE REVIEW

**Comment by tnozicka** [comment]
/override ci/prow/verify-commits
(a rebase will never pass)

**Comment by tnozicka** [comment]
current status: bootstrap failing with
```
Error: [service-account-issuer is a required flag, --service-account-signing-key-file and --service-account-issuer are required flags]
```

**Comment by tnozicka** [comment]
/retest
(gcp issues)

**Comment by tnozicka** [comment]
/test e2e-aws-serial
https://github.com/openshift/origin/pull/25717 merged

**Comment by tnozicka** [comment]
https://github.com/openshift/origin/pull/25731 and https://github.com/openshift/origin/pull/25730 merged
/retest

**Comment by soltysh** [comment]
@tnozicka the commits looks good, although I'd probably split `UPSTREAM: <carry>: Add OpenShift tooling, images, configs and docs` into separate elements, for sure the watch-termination bit would be better of in a separate commit.

**Comment by tnozicka** [comment]
> @tnozicka the commits looks good, although I'd probably split UPSTREAM: <carry>: Add OpenShift tooling, images, configs and docs into separate elements, for sure the watch-termination bit would be better of in a separate commit.

@soltysh I haven't squashed yet, this is from the previous rebase. But I believe we want to keep our tooling want to keep it in one commit so it's easier to pick and you don't have to solve conflicts N times. iirc there were also several commits modifying the same file back and forth.

**Comment by tnozicka** [comment]
gcp flakes :(
/retest

File: openshift-hack/e2e/annotate/rules.go [hunk 1]
```diff
@@ -57,7 +58,7 @@ var (
 			`ServiceAccounts should ensure a single API token exists`,   // We create lots of secrets
 			`unchanging, static URL paths for kubernetes api services`,  // the test needs to exclude URLs that are not part of conformance (/logs)
 			`Services should be able to up and down services`,           // we don't have wget installed on nodes
-			`Network should set TCP CLOSE_WAIT timeout`,                 // possibly some difference between ubuntu and fedora
+			`KubeProxy should set TCP CLOSE_WAIT timeout`,               // possibly some difference between ubuntu and fedora
```
Comment (aojea) [inline openshift-hack/e2e/annotate/rules.go:61]: we should update the comment, is not accurate 
```
the test require communication to port 11302 in the cluster nodes
```

**Comment by tnozicka** [comment]
rpm server flake
/retest

**Comment by tnozicka** [comment]
rpm server flake again
/retest

**Comment by tnozicka** [comment]
upgrade jobs look like flakes
/retest

**Comment by tnozicka** [comment]
ci-operator flakes
/retest

**Comment by wking** [comment]
openshift/cluster-version-operator#489 landed, bringing in 1.20 clients there.

/retest

**Comment by tnozicka** [comment]
CVO still shows 
```
2020-12-12T07:12:16.254234691Z E1212 07:12:16.254181       1 task.go:81] error running apply for flowschema "openshift-etcd-operator" (72 of 670): no kind "FlowSchema" is registered for version "flowcontrol.apiserver.k8s.io/v1beta1" in scheme "k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go:30"
```

/retest
?

**Comment by vrutkovs** [comment]
Seems updated CVO is pulled, no flowschema errors

/retest

**Comment by tnozicka** [comment]
> Seems updated CVO is pulled, no flowschema errors

looking at the latest self-upgrade job
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_kubernetes/471/pull-ci-openshift-kubernetes-master-e2e-aws-selfupgrade/1338219170780680192/artifacts/e2e-aws-selfupgrade/gather-must-gather/
I still see the error for CVO
```
2020-12-13T23:39:07.178086966Z E1213 23:39:07.178055       1 task.go:81] error running apply for flowschema "openshift-etcd-operator" (72 of 670): no kind "FlowSchema" is registered for version "flowcontrol.apiserver.k8s.io/v1beta1" in scheme "k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go:30"
```

**Comment by tnozicka** [comment]
I see @sttts already figured it out, great job btw. , going to pick https://github.com/openshift/kubernetes/pull/485/commits/f1a50db6186a806e965bb449011b456f18255ecf

**Comment by tnozicka** [comment]
infra flake `Error setting IAM policy`
/retest

**Comment by tnozicka** [comment]
build test fix in https://github.com/openshift/origin/pull/25761

selfupgrade paniced with
```
pods/openshift-monitoring_prometheus-adapter-66d5b468c5-f6kmf_prometheus-adapter.log.gz:E1214 09:33:32.333299       1 runtime.go:78] Observed a panic: &errors.errorString{s:"killing connection/stream because serving request timed out and response had been started"} (killing connection/stream because serving request timed out and response had been started)
```

/retest

**Comment by tnozicka** [comment]
```
2020/12/14 10:16:42 Executing pod "e2e-agnostic-cmd-ipi-conf-gcp"
2020/12/14 10:16:47 error: could not wait for pod 'e2e-agnostic-cmd-ipi-conf-gcp': it is no longer present on the cluster (usually a result of a race or resource pressure. re-running the job should help)
2020/12/14 10:16:47 error: could not wait for pod 'e2e-agnostic-cmd-ipi-conf-gcp': it is no longer present on the cluster (usually a result of a race or resource pressure. re-running the job should help)
time="2020-12-14T10:16:47Z" level=error msg="Waiting for container artifacts in pod e2e-agnostic-cmd-ipi-conf-gcp in namespace ci-op-fqkk9flk" error="pods \"e2e-agnostic-cmd-ipi-conf-gcp\" not found"
```
/retest

**Comment by tnozicka** [comment]
ci infra down
`error: build error: failed to pull image: After retrying 2 times, Pull image still failed due to error: unable to retrieve auth token: invalid username/password: unauthorized: unable to validate token: Timeout`
/retest

**Comment by soltysh** [comment]
Looks like google's up, let's try
/test e2e-gcp

**Comment by tnozicka** [comment]
the build test fix for fips also merged
/retest

**Comment by tnozicka** [comment]
gcp infra flake
/test ci/prow/e2e-gcp

**Comment by tnozicka** [comment]
/retest
to have more data on selfupgrade but so far it is OOM and prometheus get's killed and the test can't talk to it

**Comment by soltysh** [comment]
/override ci/prow/verify-commits 
/override ci/prow/e2e-gcp 

The first is known to fail on k8s bump, the 2nd is limited to 4 failures, 2 of sig-apimachinery are expected, will be fixed after bump in origin, one for build is already merged, last for networking will be BZ-ed

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by soltysh** [comment]
/override ci/prow/e2e-aws-fips

the failures are expected and explained above

**Pull request merged**